File: src/main/webapp/js/flashinfo.js
/*
 * Copyright © 2016 ESUP-Portail (https://www.esup-portail.org/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var flashInfoPortlet = flashInfoPortlet || {};


flashInfoPortlet.init = function($, namespace, portletId, openKnownMoreInModal) {


    (function initContainer($, namespace, portletId, openKnownMoreInModal) {
        $(window).bind('load', function () {

            //console.log("namespace : " + namespace);
            if ($(namespace + ' #myCarousel_' + portletId + " .item").length > 0) {
                var defaults      = {
                    selector:          namespace + ' #myCarousel_' + portletId + " .item img"   ,
                    parent:               namespace + ' #myCarousel_' + portletId + " .item",
                    exclude:              [ 'rgb(0,0,0)', 'rgb(255,255,255)' ],
                    normalizeTextColor:   false,
                    normalizedTextColors:  {
                        light:      "#fff",
                        dark:       "#000"
                    },
                    lumaClasses:  {
                        light:      "ab-light",
                        dark:       "ab-dark"
                    }
                };
                //$.adaptiveBackground.run(defaults);

                // évènement de fin de recherche de couleur dominante d'une image
                $(namespace + ' #myCarousel_' + portletId + " .item img").on('ab-color-found', function(ev,payload) {
                    //console.log(payload.color);   // The dominant color in the image.
                    //console.log(payload.palette); // The color palette found in the image.
                    //console.log(ev);   // The jQuery.Event object
                    colorCarouselElements($, namespace, portletId);
                });

                 // évènement de slide du carousel terminé
                $(namespace + ' #myCarousel_' + portletId).on('slid.bs.carousel', function (e) {
                    colorCarouselElements($, namespace, portletId);
                });

                // event on open flash know more url
                if (openKnownMoreInModal) {
                    $(namespace + ' #myCarousel_' + portletId + ' .carousel-caption>a').on('click', function (e) {
                        e.preventDefault();
                        $(namespace + ' #flashInfoModal' + portletId).modal('show').find('.modal-body').load($(this).attr('href'));
                    });
                }
            } else {
                console.log("FlashInfo doesn't have element to show !")
            }
        });

    })($, namespace, portletId, openKnownMoreInModal);

    function colorCarouselElements($, namespace, portletId) {
        var selector = $(namespace + ' #myCarousel_' + portletId + ' .item.active');
        if (selector && selector.length > 0) {
            if(selector.hasClass("ab-dark")) {
                $(namespace + ' #myCarousel_' + portletId).removeClass("ab-light").addClass("ab-dark");
            } else {
                $(namespace + ' #myCarousel_' + portletId).removeClass("ab-dark").addClass("ab-light");
            }
        }
    }
};

File: src/main/java/org/esco/portlet/flashinfo/model/FlashInfo.java
/**
 * Copyright © 2016 ESUP-Portail (https://www.esup-portail.org/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.esco.portlet.flashinfo.model;

import java.io.Serializable;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class FlashInfo implements Serializable {

	private String imgLink;
	private String title;
	private String text;
	private String knowMoreLink;
	private String imgAlt;
	private Set<Rubrique> rubriques;

	public FlashInfo() {
		super();
	}

	@JsonCreator
	public FlashInfo(@JsonProperty(value = "mediaUrl", required = true) String imgLink, @JsonProperty(value = "title", required = true) String title,
					 @JsonProperty(value = "summary", required = true) String text, @JsonProperty(value = "link", required = true) String knowMoreLink,
					 @JsonProperty("rubriques") Set<Rubrique> rubriques) {
		this.imgLink = imgLink;
		this.title = title;
		this.text = text;
		this.knowMoreLink = knowMoreLink;
		this.rubriques = rubriques;
	}

	public FlashInfo(String imgL, String title, String text, String kmL, String alt){
		super();
		this.imgLink = imgL;
		this.title = title;
		this.text = text;
		this.knowMoreLink = kmL;
		this.imgAlt = alt;
	}

	public String getImgLink() {
		return imgLink;
	}

	public void setImgLink(String imgLink) {
		this.imgLink = imgLink;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public String getKnowMoreLink() {
		return knowMoreLink;
	}

	public void setKnowMoreLink(String knowMoreLink) {
		this.knowMoreLink = knowMoreLink;
	}

	public String getImgAlt() {
		return imgAlt;
	}

	public void setImgAlt(String imgAlt) {
		this.imgAlt = imgAlt;
	}

	public Set<Rubrique> getRubriques() {
		return rubriques;
	}

	public void setRubriques(final Set<Rubrique> rubriques) {
		this.rubriques = rubriques;
	}

	@Override
	public String toString() {
		return "FlashInfo{" +
				"imgLink='" + imgLink + '\'' +
				", title='" + title + '\'' +
				", text='" + text + '\'' +
				", knowMoreLink='" + knowMoreLink + '\'' +
				", imgAlt='" + imgAlt + '\'' +
				", rubriques='" + rubriques.toString() + '\'' +
				'}';
	}
}
